### PR TITLE
Fixed KdocComments

### PR DESCRIPTION
**Pull request by DrAlexD**
### What's done:
- Changed warning `KDOC_NO_CONSTRUCTOR_PROPERTY` to `KDOC_NO_CONSTRUCTOR_PROPERTY_WITH_COMMENT` in some cases when property or parameter has comment before.
- Added check is warning `KDOC_NO_CONSTRUCTOR_PROPERTY_WITH_COMMENT` fixable for case when property or parameter has any tags in KDoc-comment before. If it has = non-fixed.
- Added configuration for `@param` tags creation: `isParamTagsForParameters` and `isParamTagsForPrivateProperties`.
- Added fixed warnings `KDOC_NO_CONSTRUCTOR_PROPERTY` and `KDOC_NO_CONSTRUCTOR_PROPERTY_WITH_COMMENT` for cases when configuration options are on, and then `private` properties and parameters must have `@param` tags in class-KDoc. Additionally added logic for replacing incorrect tag with correct one regardless of whether configuration is enabled or not.
- Added non-fixed warning `KDOC_EXTRA_PROPERTY` for redundant `@param` tags in class-KDoc.
- Reworked fix tests. Added tests for cases when `@property` or `@param` tag exist in class-KDoc.
- Added new warning tests.

It's part of #1737

### CODE REVIEW

File: diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunnerFactory.kt [hunk 1]
```diff
@@ -5,14 +5,20 @@ import com.saveourtool.diktat.api.DiktatBaselineFactory
...
 import com.saveourtool.diktat.api.DiktatRuleConfigReader
+import com.saveourtool.diktat.api.DiktatRuleSet
 import com.saveourtool.diktat.api.DiktatRuleSetFactory
 import java.io.OutputStream
 import java.nio.file.Path
 
 /**
  * A factory to create [DiktatRunner]
  *
+ * @property diktatRuleConfigReader a reader for [DiktatRuleConfig]
```
Comment (nulls) [inline diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunnerFactory.kt:18]: actually, read https://stackoverflow.com/questions/45032436/what-is-the-difference-between-properties-and-parameters-in-kotlin one and looks line the private property is `@param`

@akuleshov7 what do you think?

Comment (nulls) [reply diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunnerFactory.kt:18]: discussed offline with @akuleshov7  -- it should be `@param`

**Review by nulls — CHANGES_REQUESTED** [review]

**Review by nulls — APPROVED** [review]

File: diktat-cli/src/main/kotlin/com/saveourtool/diktat/cli/DiktatProperties.kt [hunk 2]
```diff
@@ -26,6 +26,9 @@ import kotlinx.cli.default
 import kotlinx.cli.vararg
 
 /**
+ * @param groupByFileInPlain
```
[the remark this answers is not in the document]
Comment (DrAlexD) [reply diktat-cli/src/main/kotlin/com/saveourtool/diktat/cli/DiktatProperties.kt:29]: That happens due to `KdocFormatting` rule and `KDOC_WRONG_TAGS_ORDER` warning. After reordering `@param` tags are always above `@property` tags.

File: diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunnerFactory.kt (hunk 1 above)
Comment (DrAlexD) [reply diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunnerFactory.kt]: Done

File: diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunner.kt [hunk 3]
```diff
@@ -15,9 +15,9 @@ private typealias RunAction = (DiktatProcessor, DiktatProcessorListener) -> Unit
 /**
  * A runner for diktat on bunch of files using baseline and reporter
  *
+ * @param diktatBaselineGenerator
```
[the remark this answers is not in the document]
Comment (DrAlexD) [reply diktat-api/src/main/kotlin/com/saveourtool/diktat/DiktatRunner.kt:18]: That happens due to KdocFormatting rule and KDOC_WRONG_TAGS_ORDER warning. After reordering @param tags are always above @property tags.

**Pull request merged**